var/spack/repos/builtin/packages/sundials/package.py

```diff
@@ -245,256 +242,173 @@ class Sundials(CMakePackage, CudaPackage, ROCmPackage):
+            args.extend([
+                define('SUNDIALS_INDEX_SIZE', intsize),
+                define('SUNDIALS_INDEX_TYPE', 'int{}_t'.format(intsize)),
+            ])
 
-        # Monitoring
         args.extend([
-            self.define_from_variant('SUNDIALS_BUILD_WITH_MONITORING', 'monitoring')
+            # Fortran interface
+            from_variant('F77_INTERFACE_ENABLE', 'fcmix'),
+            from_variant('F2003_INTERFACE_ENABLE', 'f2003'),
+            # library type
+            from_variant('BUILD_SHARED_LIBS', 'shared'),
+            from_variant('BUILD_STATIC_LIBS', 'static'),
+            # Generic (std-c) math libraries
+            from_variant('USE_GENERIC_MATH', 'generic-math'),
+            # Monitoring
+            from_variant('SUNDIALS_BUILD_WITH_MONITORING', 'monitoring'),
+            # Profiling
+            from_variant('SUNDIALS_BUILD_WITH_PROFILING', 'profiling'),
+            from_variant('ENABLE_CALIPER', 'caliper'),
         ])
 
-        # Profiling
-        args.extend([
-            self.define_from_variant('SUNDIALS_BUILD_WITH_PROFILING', 'profiling')
-        ])
-        if '+profiling+caliper' in spec:
-            args.extend([
-                '-DENABLE_CALIPER=ON',
-                '-DCALIPER_DIR=%s' % spec['caliper'].prefix
-            ])
+        if '+caliper' in spec:
+            args.append(define('CALIPER_DIR', spec['caliper'].prefix))
 
         # parallelism
         args.extend([
-            self.define_from_variant('MPI_ENABLE', 'mpi'),
-            self.define_from_variant('OPENMP_ENABLE', 'openmp'),
-            self.define_from_variant('PTHREAD_ENABLE', 'pthread'),
-            self.define_from_variant('ENABLE_SYCL', 'sycl')
+            from_variant('MPI_ENABLE', 'mpi'),
+            from_variant('OPENMP_ENABLE', 'openmp'),
+            from_variant('PTHREAD_ENABLE', 'pthread'),
+            from_variant('ENABLE_SYCL', 'sycl'),
+            from_variant('CUDA_ENABLE', 'cuda'),
+            from_variant('ENABLE_HIP', 'rocm'),
+            from_variant('HYPRE_ENABLE', 'hypre'),
+            from_variant('KLU_ENABLE', 'klu'),
+            from_variant('LAPACK_ENABLE', 'lapack'),
+            from_variant('PETSC_ENABLE', 'petsc'),
+            from_variant('RAJA_ENABLE', 'raja'),
+            from_variant('SUPERLUMT_ENABLE', 'superlu-mt'),
+            from_variant('SUPERLUDIST_ENABLE', 'superlu-dist'),
+            from_variant('Trilinos_ENABLE', 'trilinos'),
+            from_variant('EXAMPLES_INSTALL', 'examples-install'),
         ])
 
         if '+cuda' in spec:
-            args.append('-DCUDA_ENABLE=ON')
-            archs = spec.variants['cuda_arch'].value
-            if archs[0] != 'none':
-                arch_str = ",".join(archs)
-                args.append('-DCMAKE_CUDA_ARCHITECTURES=%s' % arch_str)
-        else:
-            args.append('-DCUDA_ENABLE=OFF')
+            args.append(define(
+                'CMAKE_CUDA_ARCHITECTURES', spec.variants['cuda_arch'].value
```

Comment:
This requires cmake 3.18 to make sense. Maybe bump the lowerbound too?

Reply:
Good suggestion, but this also opens up a can of worms about "minimum cmake version" and the history of a repo. There are other CMake features besides CUDA support that require different versions -- and CMake is so good with backward compatibility that I can't see any justification for ever building `package ^cmake@old`.

Reply:
We handle the case where the CMake version is < 3.18.0 but `CMAKE_CUDA_ARCHITECTURES` has been provied in SUNDIALS, so no need to bump the minimum version.